xkbcommon/xkb.py

```diff
@@ -143,17 +143,7 @@ def __init__(self, no_default_includes=False, no_environment_names=False):
         context = lib.xkb_context_new(flags)
         if not context:
             raise XKBError("Couldn't create XKB context")
-        # This nasty hack is necessary to keep "lib" around long
-        # enough for us to use it to free the context while python is
-        # shutting down.  The obvious way of writing it works fine on
-        # python3; this version is necessary for compatibility with
-        # python2.
-        def free_context_closure():
-            saved_lib = lib
-            def free_context(obj):
-                saved_lib.xkb_context_unref(obj)
-            return free_context
-        self._context = ffi.gc(context, free_context_closure())
+        self._context = ffi.gc(context, lib.xkb_context_unref)
```

Comment:
I'm afraid my comment wasn't explicit enough!  `lib` being unloaded while references to its symbols are still around is still a problem under python3, but the solution is simpler.  See how I solved it in cairocffi here: https://github.com/Kozea/cairocffi/pull/79

Reply:
That is interesting, I hadn't known this. Thanks for the pointer, I changed this over to follow that same convention.